e2e/tests/providers.rs

```diff
@@ -1011,6 +1012,90 @@ async fn test_build_with_provider() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn send_transaction_and_await_status() -> Result<()> {
+    let wallet = launch_provider_and_get_wallet().await?;
+    let provider = wallet.provider();
+
+    let consensus_parameters = provider.consensus_parameters().await?;
+    let inputs = wallet
+        .get_asset_inputs_for_amount(*consensus_parameters.base_asset_id(), 100, None)
+        .await?;
+    let outputs = wallet.get_asset_outputs_for_amount(
+        &Bech32Address::default(),
+        *consensus_parameters.base_asset_id(),
+        100,
+    );
+
+    // Given
+    let mut tb = ScriptTransactionBuilder::prepare_transfer(inputs, outputs, TxPolicies::default());
+    wallet.add_witnesses(&mut tb)?;
+
+    let tx = tb.build(provider).await?;
+
+    // When
+    let status = provider.send_transaction_and_await_status(tx, true).await?;
+
+    // Then
+    assert_eq!(status.len(), 3);
+    assert!(status.iter().enumerate().all(|(i, tx_status)| {
+        matches!(
+            (i, tx_status.clone().unwrap()),
+            (0, TxStatus::Submitted { .. })
+                | (1, TxStatus::PreconfirmationSuccess { .. })
+                | (2, TxStatus::Success { .. })
+        )
+    }));
+    Ok(())
+}
+
+#[tokio::test]
+async fn send_transaction_and_subscribe_status() -> Result<()> {
+    let wallet = launch_provider_and_get_wallet().await?;
+    let provider = wallet.provider();
```

Comment:
Maybe we can eliminate the falkyness with a custom block production interval?
```suggestion
     let block_time = 3u32;
    let config = NodeConfig {
        block_production: Trigger::Interval {
            block_time: std::time::Duration::from_secs(block_time.into()),
        },
        ..NodeConfig::default()
    };
    let wallets =
        launch_custom_provider_and_get_wallets(WalletsConfig::default(), Some(config), None)
            .await?;
    let wallet = &wallets[0];
    let provider = wallet.provider();
```

Reply:
Done.